flink-test-utils-parent/flink-connector-testing/src/test/java/org/apache/flink/connectors/test/common/utils/TestDataMatchersTest.java

```diff
@@ -140,7 +175,22 @@ public void testOutOfOrder() throws Exception {
             final List<String> result = unionLists(splitA, splitB, reverted);
             final TestDataMatchers.MultipleSplitDataMatcher<String> matcher =
                     TestDataMatchers.matchesMultipleSplitTestData(testDataCollection);
-            assertMatcherFailedWithDescription(result.iterator(), matcher, "Unexpected record '3'");
+            String expectedDescription =
+                    "Unexpected record '3' at position 6\n"
+                            + "Current progress of multiple split test data validation:\n"
+                            + "Split 0 (3/3): \n"
+                            + "alpha\n"
+                            + "beta\n"
+                            + "gamma\n"
+                            + "Split 1 (3/3): \n"
+                            + "one\n"
+                            + "two\n"
+                            + "three\n"
+                            + "Split 2 (0/3): \n"
+                            + "1\t<----\n"
+                            + "2\n"
+                            + "3\n";
+            assertMatcherFailedWithDescription(result.iterator(), matcher, expectedDescription);
```

Comment:
Here it would be nice to also see the actual elements and not just the expected. For example, did we just receive incorrect output or was it swapped?

Reply:
Thanks for the review @AHeise ! I appended the remaining received elements to the mismatchDescription in my latest push so that it can provide more info for debugging.